test/test_genome.py

```diff
@@ -346,89 +347,118 @@ def test_is_gene_in_output_region(rng_seed):
+    acceptable_error_interval = 0.05  # Todo: get comment for this value from Jakob and Max
+    n = 1000
+    n_immutable_genes = count_n_immutable_genes(n_input, n_output, n_row)
+    n_mutations_expected = n * mutation_rate * (len(genome.dna) - n_immutable_genes)
+    print(len(genome.dna), n_immutable_genes)
+    n_mutations = 0
+    for _ in range(n):
+        temp_dna = genome.dna
+        genome.mutate(mutation_rate, rng)
+        current_n_mutations = count_mutations(temp_dna, genome.dna)
+        n_mutations += current_n_mutations
+
+    assert abs(n_mutations_expected - n_mutations) < (
+        n_mutations_expected * acceptable_error_interval
+    )
 
-    # mutating function gene in active region returns False
-    assert genome._mutate_hidden_region(list(dna), 6, active_regions, rng) is False
-    # mutating inactive genes in active region returns True
-    assert genome._mutate_hidden_region(list(dna), 7, active_regions, rng) is True
-    assert genome._mutate_hidden_region(list(dna), 8, active_regions, rng) is True
 
-    # mutating any gene in active region without silent genes returns False
-    assert genome._mutate_hidden_region(list(dna), 9, active_regions, rng) is False
-    assert genome._mutate_hidden_region(list(dna), 10, active_regions, rng) is False
-    assert genome._mutate_hidden_region(list(dna), 11, active_regions, rng) is False
+def test_only_silent_mutations(genome_params, rng_seed):
+    genome = cgp.Genome(**genome_params)
+    rng = np.random.RandomState(rng_seed)
+    genome.randomize(rng)
 
+    only_silent_mutations_0 = genome.mutate(mutation_rate=0, rng=rng)
+    assert only_silent_mutations_0
 
-def test_mutate_output_region(rng_seed):
-    rng = np.random.RandomState(rng_seed)
-    genome = cgp.Genome(1, 1, 2, 1, None, (cgp.Add,))
-    dna = [
-        ID_INPUT_NODE,
-        ID_NON_CODING_GENE,
-        ID_NON_CODING_GENE,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        ID_OUTPUT_NODE,
+    only_silent_mutations_1 = genome.mutate(mutation_rate=1, rng=rng)
+    assert not only_silent_mutations_1
+
+
+# Overwrite global genome_params() from conftest,
+# since tests below strongly depend on these variable values
+@fixture
+def genome_params():
+    return {
+        "n_inputs": 2,
+        "n_outputs": 1,
+        "n_columns": 3,
+        "n_rows": 3,
+        "levels_back": 2,
+        "primitives": (cgp.Add, cgp.Sub, cgp.ConstantFloat),
+    }
+
+
+def test_permissible_values_hidden(genome_params, rng_seed):
+    genome = cgp.Genome(**genome_params)
+
+    # test function gene
+    gene_idx = 6  # function gene of first hidden gene
+    gene = 0  #
+    region_idx = 1000  # can be any number, since not touched for function gene
+    permissible_function_gene_values = genome._determine_permissible_values_hidden(
+        gene_idx, gene, region_idx
+    )
+
+    # test input gene
+    gene_idx = 16  # first input gene in second hidden layer
+    gene = 0
+    region_idx = 5
```

Comment:
can't you compute this from the gene idx?

Reply:
eg: `region_idx = int(gene_idx / (genome.primitives.max_arity + 1)) ` ? (+1 because of the function gene)

Reply:
yep! you could also use the floor division operator (`//`) directly instead of casting: `gene_idx // (genome.primitives.max_arity + 1)`